tests/bluetooth/test_service_emulation.py

```diff
@@ -0,0 +1,170 @@
+#  Copyright 2025 Google LLC.
+#  Copyright (c) Microsoft Corporation.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+import pytest
+import pytest_asyncio
+from test_helpers import execute_command
+
+from . import (BATTERY_SERVICE_UUID, FAKE_DEVICE_ADDRESS,
+               HEART_RATE_SERVICE_UUID, create_gatt_connection,
+               disable_simulation, setup_device, setup_granted_device)
+
+
+async def simulate_service(websocket, context_id: str, address: str, uuid: str,
+                           type: str) -> None:
+    await execute_command(
+        websocket, {
+            'method': 'bluetooth.simulateService',
+            'params': {
+                'context': context_id,
+                'address': address,
+                'uuid': uuid,
+                'type': type
+            }
+        })
+
+
+async def check_services(websocket, context_id: str,
```

Comment:
nit: same here. Consider making it "get_services"

Reply:
Done.